impress-org/givewp-next-gen
# Refactor: Update Terms & Conditions block ```showInForm``` display

## Description

This PR updates the "In form" agreement text display style for the Terms and conditions block. This display type originally was setup similar to the original agreement text for v2 forms, in where a user would need to open/reveal the agreement by clicking a show terms button. We have updated this by removing the "show terms" button and displaying the agreement text initially without the need to open/reveal it.

Additional updates include: 
- Includes new class to handle post request data for DesignPreview. This is required to allow the html attributes using kses_post(). So that the provided html tags for the Consent field & any other html field render properly in DesignPreview
- Fix for applying the global attributes to the block correctly by passing the global settings through the window object
- Updated default agreement text value to include a paragraph tag - replicates initial spacing for Interweave
- Improved styling
- Update ```Checkbox``` function name to ```CheckboxPlaceholder``` to improve readabillity

## Affects

Terms & Conditions block

## Visuals

<img width="1723" alt="Screen Shot 2023-08-02 at 9 55 13 AM" src="https://github.com/impress-org/givewp-next-gen/assets/75056371/04a54bcd-6a4e-46e3-83bd-b7befd86037d">
<img width="1723" alt="Screen Shot 2023-08-02 at 9 56 29 AM" src="https://github.com/impress-org/givewp-next-gen/assets/75056371/70496271-00bf-4894-99af-cc963fe9cf06">

## Testing Instructions

- Verify global settings are used correctly upon block creation.
- Select customize show terms in form & use the editor to edit the text.
- View on Design Preview
- View on Frontend

## Pre-review Checklist






---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1204525695989154
